### ISSUE TITLE
Sweep: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#13]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 It seems that you are wishing to name your model weights file using current epoch, validation loss and wandb (Weights & Biases) run name. Current implementation does allow using epoch and validation loss for filename but doesn't support wandb run name out-of-the-box.
Here's a workaround that could help you in achieving what you want. You can build a custom callback to format the filename string. This is achievable via PyTorch Lightning Callback system since it provides hooks to many important events that occur during the training or evaluation.
Here's an illustrative example:
```python
from pytorch_lightning import callbacks
class CheckpointEveryNSteps(callbacks.ModelCheckpoint):
    def __init__(self, save_step_frequency, prefix="NStep", *args, **kwargs):
        """
        Save the model every `save_step_frequency` steps
        """
        super().__init__(*args, **kwargs)
        self.save_step_frequency = save_step_frequency
        self.prefix = prefix
        
    def _save_model(self, trainer, pl_module, ckpt_name_metrics):
        # Use your own method to extract wandb run name. Let's say it's `wandb_run_name`
        epoch = trainer.current_epoch
        loss = ckpt_name_metrics['val_loss']
        filename = f"{self.prefix}_{wandb_run_name}_epoch{epoch}_loss{loss:.2f}.ckpt"
        trainer.save_checkpoint(os.path.join(self.dirpath, filename))
    def on_batch_end(self, trainer, pl_module):
        # Check if it's time to save model
        steps = trainer.global_step
        if steps % self.save_step_frequency == 0:
            self._save_model(trainer, pl_module, trainer.logged_metrics)
checkpoint_callback = CheckpointEveryNSteps(save_step_frequency=500, monitor='val_loss')
trainer = Trainer(callbacks=[checkpoint_callback])
```
Above code should get you what you want. However, please be sure to modify it as per your needs since it's a rough sketch of your solution. As per this code, a checkpoint will be saved after every `save_step_frequency` steps are completed. Tracked metrics can be logged and accessed using `trainer.logged_metrics`. `wandb_run_name` should be your method of extracting run name from wandb.

<details open>
<summary>Checklist</summary>

- [X] Modify `src/lightning/pytorch/callbacks/model_checkpoint.py` ✓ https://github.com/ayulockin/pytorch-lightning/commit/75507732e0bcc1081cb66d601dbe3f56b43b8a5b [Edit](https://github.com/ayulockin/pytorch-lightning/edit/sweep/format_wandb_run_name_in_pytorch_checkpo/src/lightning/pytorch/callbacks/model_checkpoint.py#L49-L204)
- [X] Running GitHub Actions for `src/lightning/pytorch/callbacks/model_checkpoint.py` ✓  [Edit](https://github.com/ayulockin/pytorch-lightning/edit/sweep/format_wandb_run_name_in_pytorch_checkpo/src/lightning/pytorch/callbacks/model_checkpoint.py#L49-L204)
- [X] Modify `tests/tests_pytorch/checkpointing/test_model_checkpoint.py` ✓ https://github.com/ayulockin/pytorch-lightning/commit/4d90e7c7855a93e938aca5ac601fd8dae8ba27cc [Edit](https://github.com/ayulockin/pytorch-lightning/edit/sweep/format_wandb_run_name_in_pytorch_checkpo/tests/tests_pytorch/checkpointing/test_model_checkpoint.py#L1-L100)
- [X] Running GitHub Actions for `tests/tests_pytorch/checkpointing/test_model_checkpoint.py` ✓  [Edit](https://github.com/ayulockin/pytorch-lightning/edit/sweep/format_wandb_run_name_in_pytorch_checkpo/tests/tests_pytorch/checkpointing/test_model_checkpoint.py#L1-L100)
</details>
